### PR TITLE
Add mock product report, helper functions

### PR DESCRIPTION
**Pull request by ryelle**
This PR starts off a "Product Report" page by adding mock data for the table. Eventually this will be pulled from a combo of 2 API endpoints, as the product revenue data endpoint won't contain product name/category/stock etc. See p7bje6-14Q-p2, "The Table"

For now, I'm pretending we've fetched all that & merged it into one array. This is then used to display a table on Analytics > Products.

This PR also sets up a few helper functions for other reports:

- The `onQueryChange` handler is moved to `nav-utils`, so table sorting & pagination can be handled in one place
- Introduces a helper function `getIdsFromQuery` to parse numeric IDs from a comma-separated string in the query (ex: `products=2,6,18` will give us the array `[ 2, 6, 18 ]`)
 
**To test**

- Go to Analytics > Products, make sure the table shows up
- Try "sorting" a table column, the URL should update with the new `order_by` (the table _won't update_ since it's not hooked up to an API)
- Go to Analytics > Revenue, make sure no functionality has changed

Note: this PR is a precursor to adding the "Compare" functionality to the table #94

### CODE REVIEW

**Review by timmyc — APPROVED** [review]
Couple of minor comments, working well and loving the mock data approach to get things started on this report <3

File: client/analytics/report/products/index.js [hunk 1]
```diff
@@ -2,22 +2,158 @@
...
+				{
+					display: orders_count,
+					value: Number( orders_count ),
+				},
+				{
+					display: 'Categories',
+					value: false,
+				},
+				{
+					display: variations.length + '',
```
[the remark this answers is not in the document]
Comment (ryelle) [reply client/analytics/report/products/index.js]: Whoops, I meant to comment about that here – right now [Table displays `null` for any falsey value](https://github.com/woocommerce/wc-admin/blob/master/client/components/table/table.js#L16), so this just converts the 0 to a string.  `numberFormat` would be a better solution 🙂

File: client/analytics/report/products/index.js [hunk 2]
```diff
@@ -2,22 +2,158 @@
...
+		const headers = this.getHeadersContent();
+
+		const tableQuery = {
+			...query,
+			orderby: query.orderby || 'date_start',
+			order: query.order || 'asc',
+		};
+		return (
+			<TableCard
+				title={ __( 'Revenue', 'wc-admin' ) }
```
[the remark this answers is not in the document]
Comment (ryelle) [reply client/analytics/report/products/index.js]: … yes 🤦🏼‍♀️

**Comment by ryelle** [comment]
Rebased & fixed the noted issues, will merge once the tests pass

**Pull request merged**